tinygrad/dtype.py

```diff
@@ -17,6 +17,10 @@ def vec(self, sz:int):
     assert sz > 1 and self.count == 1, f"can't vectorize {self} with size {sz}"
     return DType(self.priority, self.itemsize*sz, f"{INVERSE_DTYPES_DICT[self.name]}{sz}", None, sz)
   def scalar(self): return DTYPES_DICT[self.name[:-len(str(self.count))]] if self.count > 1 else self
+  def __rmod__(self, other: Scalar) -> Scalar:
```

Comment:
I don't think you actually want to mod by the dtype! This logic should be in exec_alu

Reply:
Perhaps mod is not correct, but I don't think the operation belongs in exec_alu because exec_alu isn't the only place that performs this adjustment business. 

__call__ is probably better because then one can do `dtypes.int8(x)` and get an "in-bounds" version of x.

Reply:
That's not better. This should not be in dtype at all.